pyoptic2/system.py

```diff
@@ -31,7 +33,8 @@ def propagate(self, source=None) :
         ri = iter(source)
         try :                
             while True :
-                r = ri.next()
+                # r = ri.next()
+                r = six.next(ri)
```

Comment:
just for interest, how has next changed?

Reply:
It's now `ri.__next__()`.